src/SAML2/XML/saml/SubjectConfirmationData.php

```diff
@@ -246,7 +246,7 @@ private function setInfo(?array $info): void
     public function addInfo(object $info): void
```

Comment:
In the other branch I've been getting rid of all adders. If we are going for immutability (even at the expense of hunge constructor signatures), then I think we shouldn't allow public methods that modify the state like this.

Reply:
I thought you wanted to keep them? Have you changed your mind?

Reply:
Yeah. I panicked first when I saw the monster signatures of some constructors, but then thought it was stupid to aim for immutability if that was going away with adders. I think we discussed this at some point, and agreed to make everything strictly immutable at least in the first iteration, and revisit it later.

Reply:
I think the aim was to make sure no illegal situations can occur.. Not per se immutability?